Sources/IO/Core/HttpDataSetReader/test/testHttpDataSetReader.js

```diff
@@ -0,0 +1,238 @@
+import test from 'tape-catch';
+
+import vtkHttpDataSetReader from 'vtk.js/Sources/IO/Core/HttpDataSetReader';
+import MockDataAccessHelper from 'vtk.js/Sources/IO/Core/DataAccessHelper/MockDataAccessHelper';
+
+function runTests(testCases) {
+  return new Promise((resolve) => {
+    // reverse test case array so that pop() returns the first test
+    testCases.reverse();
+
+    // definition of test runner method
+    // -> Runs all test cases until none are left, or an error occurs
+    const processNextTestCase = () => {
+      const testCase = testCases.pop();
+      if (!testCase) {
+        resolve();
+        return;
+      }
+
+      new Promise(testCase).then(
+        (_) => {
+          processNextTestCase();
+        },
+        (err) => {
+          console.error(err);
+        }
+      );
+    };
+
+    // start test sequence with first test
+    processNextTestCase();
+  });
```

Comment:
This can be simplified to:
```suggestion
  let promise = Promise.resolve();
  testCases.forEach((testCase) => {
    promise = promise.then(() => new Promise(testCase));
  });
  promise.catch((err) => {
    console.error(err);
  });
  return promise;
```

Reply:
This is more of a readability change than a functional change.